rust/src/pop3/pop3.rs

```diff
@@ -132,10 +133,18 @@ use nom7::{
     IResult,
 };
 
-fn parse_unknown_message(input: &[u8]) -> IResult<&[u8], &[u8]> {
-    let (input, value) = take_until("\r\n")(input)?;
+#[inline]
+fn is_base64_char(b: u8) -> bool {
+    nom7::character::is_alphanumeric(b) || b"+/=".contains(&b)
+}
+
+fn parse_base64_message(input: &[u8]) -> IResult<&[u8], String> {
+    let (input, value) = nom7::combinator::map_res(
+        nom7::bytes::complete::take_while(is_base64_char),
+        std::str::from_utf8,
```

Comment:
We do not need to do utf8 validation + string allocation as the result of `parse_base64_message` is ignored anyways (`_base64_blob`)